cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/data/ClientConfig.java

```diff
@@ -138,7 +138,7 @@ private ClearSnapshotStrategy parseClearSnapshotStrategy(boolean hasDeprecatedOp
                                                              boolean clearSnapshot,
                                                              String clearSnapshotStrategyOption)
     {
-        if (hasDeprecatedOption)
+        if (hasDeprecatedOption && clearSnapshotStrategyOption == null)
```

Comment:
If the deprecated option is set, the warning should always be logged, regardless whether the new option is present or not. 

```java
if (hasDeprecatedOption)
{
  log warning
  if (clearSnapshotStrategyOption == null)
  {
    return the default based on the value
  }
}    
```

Reply:
Makes sense, will update it